lib/specValidator.js

```diff
@@ -42,6 +45,130 @@ class SpecValidator {
+      });
+    } else {
+      return self.findObject(refObj.localReference.accessorProperty, self.specInJson, refObj.localReference.value, self.specPath);
+    }
+  }
+
+  mergeResolvedAllOfObjects(source, target) {
+    let self = this;
+    if (!source || (source && typeof source !== 'object')) {
+      return Promise.reject(new Error(`source must be of type "object".`));
+    }
+    if (!target || (target && typeof target !== 'object')) {
+      return Promise.reject(new Error(`target must be of type "object".`));
+    }
+    //merge the target model's properties
+    source.properties = utils.merge(source.properties, target.properties);
+    //merge the array of required properties
+    if (target.required) {
+      if (!source.required) {
+        source.required = [];
+      }
+      source.required = [...new Set([...source.required, ...target.required])];
+    }
+    //merge x-ms-azure-resource
+    if (target['x-ms-azure-resource']) {
+      source['x-ms-azure-resource'] = target['x-ms-azure-resource'];
+    }
+    return Promise.resolve(source);
+  }
+
+  getRefDefinitionInAllOf(item) {
+    let self = this;
+    if (!item || (item && typeof item !== 'object')) {
+      return Promise.reject(new Error(`item must be of type "object".`));
+    }
+    if (item['$ref']) {
+      return Promise.resolve(self.getDefinitionFromReference(item['$ref']));
+    } else {
+      return Promise.resolve(item);
+    }
+  }
+
+  composeAllOf(model, item) {
+    let self = this;
+    return self.getRefDefinitionInAllOf(item).then(function(result) {
+      if (result && result.allOf) {
+        return self.resolveAllOf(result).then(function(res) {
+          return Promise.resolve(self.mergeResolvedAllOfObjects(model, result));
+        }).catch(function (err) {
+          log.error(err);
+          return Promise.reject(err);
+        });
+      } else {
+        return Promise.resolve(self.mergeResolvedAllOfObjects(model, result));
+      }
+    });
+  }
+
+  resolveAllOf(model) {
+    let self = this;
+    if (!model || (model && typeof model !== 'object')) {
+      return Promise.reject(new Error(`model cannot be null or undefined and must of type object.`));
+    }
+    if (model.allOf) {
+      let allOfItemPromiseFactories = model.allOf.map(function(item) {
+        return Promise.resolve(self.composeAllOf(model, item));
+      });
+      return utils.executePromisesSequentially(allOfItemPromiseFactories);
+    }
+    return Promise.resolve(model);
+  }
+
+  makeModelsStricter() {
+    let self = this;
+    let spec = self.specInJson;
+    let definitions = spec.definitions;
+    let modelNames = Object.keys(self.specInJson.definitions);
+    let modelPromiseFactories = modelNames.map(function (modelName) {
+      let model = definitions[modelName];
+      if (model && !model.additionalProperties) {
+        model.additionalProperties = false;
+      }
+      return self.resolveAllOf(model);
```

Comment:
resolved the above comments in the new commit. 
Added a new class `SpecResolver` that does the grunt work related to resolving file references and allof(s) and setting additionalProperties to false (if not set to true). As discussed this is done in a separate step.